lib/inc/internal/tokenizer.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "tokens.hpp"
+
+#include <boost/nowide/fstream.hpp>
+#include <vector>
+#include <queue>
+#include <string>
+
+namespace hocon {
+
+    // This exception should not leave this file
+    class problem_exception : std::runtime_error {
+    public:
+        problem_exception(problem prob);
+        problem const& get_problem() const;
+
+    private:
+        problem _problem;
+    };
+
+    class config_exception : public std::runtime_error {
```

Comment:
Looking at the ruby port, it looks like we might want `config_exception` available everywhere so we can extend it with things like `config_bug_or_broken_error`. Do you think we should break this out into it's own `config_error.hpp`? I actually have need of it in the `abstract_config_node` class too. (I could also be totally wrong and we're doing it some other way)